### PR TITLE
jobop(engine): cancel a job immediately when it is failover

### PR DESCRIPTION
**Pull request by amyangfei**
### What problem does this PR solve?

Issue Number: ref #6749

### What is changed and how it works?

If a job is being canceled, don't re-create it and terminate it immediately (during iterating pending jobs).

#### Tests 

 - Unit test
 - Integration test

### Release note 

```release-note
None
```

### CODE REVIEW

File: engine/servermaster/jobmanager_test.go [hunk 1]
```diff
@@ -712,12 +715,14 @@ func TestJobManagerIterPendingJobs(t *testing.T) {
...
 
 	// job is being backoff
+	mockJobOperator.EXPECT().IsJobCanceling(ctx, jobID).Times(1).Return(false)
 	mockBackoffMgr.EXPECT().Terminate(jobID).Times(1).Return(false)
 	mockBackoffMgr.EXPECT().Allow(jobID).Times(1).Return(false)
 	err = mgr.Tick(ctx)
 	require.NoError(t, err)
 
 	// job will be terminated
+	mockJobOperator.EXPECT().IsJobCanceling(ctx, jobID).Times(1).Return(false)
```
Comment (gozssky) [inline engine/servermaster/jobmanager_test.go:725]: It seems the test does not cover the `true` branch.

Comment (amyangfei) [reply engine/servermaster/jobmanager_test.go]: fixed, add one more test.

**Review by gozssky — APPROVED** [review]

**Review by CharlesCheung96 — APPROVED** [review]

**Pull request merged**